File: server/Models/Vault.cs
using System.ComponentModel.DataAnnotations;

namespace keeper_final.Models;

public class Vault : RepoItem<int>
{
  public string Name { get; set; }
  public string Description { get; set; }
  [Url] public string Img { get; set; }
  public bool? IsPrivate { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Vault.js
import { Profile } from "./Account.js";
import { DataBaseItem } from "./DataBaseItem.js";

export class Vault extends DataBaseItem{

  constructor(data){
    super(data)
    this.name = data.name
    this.description = data.description
    this.imgUrl = data.img
    this.isPrivate = data.isPrivate
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
  }


}